File: challenging/11_movie_list_fetch/movie_list_fetch.go
package main

import (
	"fmt"
	"io/ioutil"
	"net/http"
)

func main() {
	resp, err := http.Get("https://ghibliapi.herokuapp.com/films")
	if err != nil {
		fmt.Println("Could not send request to API:", err)
		return
	}
	defer resp.Body.Close()
	if resp.StatusCode != 200 {
		fmt.Println("Expected status code 200 but received", resp.StatusCode)
		return
	}
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println("Could not read the response body:", err)
		return
	}
	fmt.Println("Processing our list of movies")
	fmt.Println(string(body))
}

File: challenging/11_movie_list_fetch/movie_list_fetch.py
import requests

def getMovieList():
    r = requests.get('https://ghibliapi.herokuapp.com/films')

    if r.status_code != 200:
        print(f'Could not send request to API: {r.status_code}')
        exit()

    print('Processing our list of movies')
    movies = r.json()
    for movie in movies:
        print(f"{movie['title']}, {movie['release_date']}")

getMovieList()